### PR TITLE
refactor composition test

### PR DESCRIPTION
**Pull request by Karsten1987**
work in progress. I tried to extract tests only for the services.

### CODE REVIEW

**Comment by Karsten1987** [comment]
So i'll narrowed it down.
The tests may only fail with connext.

linux wo/ connext: [Build Status](http://ci.ros2.org/job/ci_linux/2102/)

I'll don't get my workspace to compile with connext, can anybody please run the executables with connext to see what's happening?

File: composition/test/composition_srv.regex [hunk 1]
```diff
@@ -0,0 +1 @@
+Incoming request: \[a: \d+, b: \d+\]
```
Comment (dirk-thomas) [inline composition/test/composition_srv.regex]: This should not check for a request but rather on the client side for a response.

File: composition/test/test_composition.py.in [hunk 2]
```diff
@@ -26,7 +26,9 @@ from launch_testing import create_handler
 def test_manual_composition():
     name = 'test_manual_composition'
     executable = '@MANUAL_COMPOSITION_EXECUTABLE@'
-    launch(name, [executable])
+    output_file = '@EXPECTED_OUTPUT_ALL@'
+    #output_file = '@EXPECTED_OUTPUT_PUBSUB@'
```
Comment (dirk-thomas) [inline composition/test/test_composition.py.in]: Can this line be removed?

File: composition/test/composition_srv.regex (hunk 1 above)
Comment (dirk-thomas) [inline composition/test/composition_srv.regex]: See b0cde6f4774d20e6cfef06402ecf4adffff95e22

File: composition/test/test_composition.py.in (hunk 2 above)
Comment (dirk-thomas) [inline composition/test/test_composition.py.in]: See ff4b2610b729ff69e00ead4fc6fa11d076c818f6

**Comment by dirk-thomas** [comment]
[Build Status](http://ci.ros2.org/job/ci_linux/2106/)

**Comment by dirk-thomas** [comment]
Ready for review.

File: composition/src/client_component.cpp [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 #include "composition/client_component.hpp"
 
+#include <inttypes.h>
```
Comment (wjwwood) [inline composition/src/client_component.cpp]: Technically should be `<cinttypes>` because this is c++ but it doesn't really matter.

Comment (dirk-thomas) [inline composition/src/client_component.cpp]: Good point. Updated.

File: composition/test/composition_srv.regex (hunk 1 above)
Comment (dirk-thomas) [reply composition/test/composition_srv.regex]: Somehow this commit has been lost in the finally merged PR. I will add it as part of ros2/build_cop#80.

**Pull request merged**